bin/compile

```diff
@@ -25,13 +25,17 @@ source ${build_pack_path}/lib/elixir_funcs.sh
 source ${build_pack_path}/lib/app_funcs.sh
 source ${build_pack_path}/lib/canonical_version.sh
 
+
+load_config
+apply_app_relative_path
+
+
 mkdir $(build_platform_tools_path)
 
 export_env_vars
 export_mix_env
 export_mix_home
 export_hex_home
-load_config
```

Comment:
Is it safe to move this? I wonder if this needs to happen after loading other things so that they can be overridden.

Reply:
In my approach, it updates `build_path` so that the major part of the script works as it is.
And since `MIX_HOME` and `HEX_HOME` rely on `build_path`, `load_config` should run before exporting them.
This is why I moved `load_config` and is the main idea of this PR.

Reply:
Sounds good! Just double-checking though, is this backward compatible or apps that do not have the new configuration setting set?

Reply:
Sure, it is backward compatible.
If user does not provide `app_relative_path` in his/her config, it works just as it was.